rosetta-source/src/main/rosetta/base-datetime-daycount-enum.rosetta

```diff
@@ -22,3 +22,30 @@ enum DayCountFractionEnum: <"The enumerated values to specify the day count frac
     RBA_BOND_BASIS_ANNUAL displayName "RBA Bond Basis (annual)" <"Per 2021 ISDA Definitions, Section 4.6.1 Day Count Fraction, paragraph (xi)(c), if Calculation Periods for the Transaction are twelve months in length (excluding any shorter or longer first and last Calculation Period), 1, except that if the first Calculation Period or the final Calculation Period is less than twelve months, Actual/Actual (ISDA) shall apply to that Calculation Period' 'per 2006 ISDA Definitions Supplement number 43, Day Count Fraction, (k)	if 'RBA Bond Basis (semi-annual)' is specified, 0.5. However, Actual/Actual (ISDA) applies to each of the first Calculation Period and the final Calculation Period if such Calculation Period is less than six months'">
     // [deprecated]
     RBA_BOND_BASIS displayName "RBA Bond Basis" <"Per 2021 ISDA Definitions, Section 4.6.1 Day Count Fraction, paragraph (xi), the calculation mechanics are driven deterministically by the Calculation Period Frequency (i.e. 0.25 if it is three months, 0.5 if it is 6 months, 1 if it is a year), except that if the first Calculation Period or the final Calculation Period is less than the Calculation Period Frequency, Actual/Actual (ISDA) shall apply to that Calculation Period">
+
+enum DateTypeRelativeToEnum: <"The enumerated values to specify the date as relative to a particular type of date defined elsewhere (1) either as an Object present in the Trade or (2) in any relevant documentation to which the Trade may refer to e.g. ISDA Definitions, Master Confirmation Agreement, etc.">
```

Comment:
- as answerd in "comment" box will remove this item from the PR
- but the same will come back as part of structuredPayout and for clarity, this is required for structuredPayout, but ti would really improve the current CDM for all types I mean the main usage is for Offset dates for being relativeTo all main types of dates you find in most products
- indeed that is a long list... per se, that is not a blocker, the point is to ensure this list is the good one - and it is, unless you find inside any date you would consider as having no business case all of them are no less than the types of dates you already find in CDM which as a matter of facts is evidencing that none of them is "exotic"
- also to answer your last question "is dateRelativeTo not sufficient ?" : no, it is not :
    - first because it is of very poor type "date" therefore when resolving an object that would have relevant "extrenalReference" value such objet can only be of type "date" that is very restricting i can see almost no usage... most date referencing will need to reference multiple dates or a type that is per se a set of attributes e.g. PaymentDates, ObservationDates, ValuationDates... these types are composite, non of them i just a stand-alone "date" which would be the only type of object that current "dateRelativeTo" would permit to produce as an output of resolving the associated "meta.externalKey
    - then because on most of the case, the date to which you want to relate does not exist ! i mean you cannot reference it because it is not yet given : use case is EXACTLY the same as the one for PayRelativeToEnum :  main rationale why an Enum makes sense is because the dates to relate to, say CalculationPeriodEndDate does not exist : that is a value that will be calculated and produced as a list of dates AFTER the implementor will have produce the schedules, based on CalculationPeriodDates paramteric valus e.g; rollDay, frequency, etc. BEFORE that when defining the trade at inception, no CalculationPeriodEndDate exis as a given date you may reference via meta.reference that is only a "concept" of a multiple values series to be produced "later" in we may say... taht is what is at stake here eventually : DateTypeRelativeToEnum is really same conceptual and business need basis as PayRelativeToEnum, have juts made it more exhaustive by covering most type of "classic" dates used by industry and as already said presetn in CDM

Reply:
Marking as resolved, since removed from the latest PR.

I suggest to bring those discussion points to the "Date Rafactoring" initiative that is on the roadmap for 2024 H2.

Reply:
The roadmap is published here and is normally maintained by the CDM SteerCo:
https://github.com/finos/common-domain-model/blob/master/ROADMAP.md

We're aware of the previous design work that has been done around dates and it will be taken as an input when that next phase starts.